content/BatchRequestContentCollection.md

```diff
@@ -0,0 +1,61 @@
+- Added requests MUST be assigned unique identifiers. Those identifiers MAY be user provided or autogenerated. The uniqueness SHOULD be enforced/validated across the same `BatchRequestContent` instance the request objects are held in AND across the `BatchRequestContentCollection` instance.
+- It MUST be possible to create dependencies between request objects. However, these dependencies SHOULD be enforced/validated across the same `BatchRequestContent` instance the request objects are held in and NOT across `BatchRequestContent` instances in the `BatchRequestContentCollection`.
+- The addition of a new request object with a dependency, SHOULD make the best effort at placing the object in the same `BatchRequestContent` instance as its dependency and throw an error in the event this is not possible.
+- It MUST be possible to configure the maximum number of request objects per `BatchRequestContent` in the `BatchRequestContentCollection` to a value lower than the default value of 20 request objects. This is to enable customized scenarios for developers and provide a means of client side optimization/throttling for workloads that may have lower maximums in batched requests.
+- It MUST possible to use a dictionary of existing unique identifiers and their corresponding to response status code to create a new/fresh `BatchRequestContentCollection` instance to unlock re-emitting/retrying failed requests from an existing `BatchRequestContentCollection` instance.
+
+> Given the similar API experience between `BatchRequestContentCollection` and `BatchRequestContent`, public documentation(code comments and examples) SHOULD encourage the use and point to `BatchRequestContentCollection` to avoid confusion for the end user. Adding new requests should automatically generate new `BatchRequestContent` instances to the `BatchRequestContentCollection`. `BatchRequestContent` should be a public facing type to enable users with customized batching scenarios.
+
+## Performance Considerations
+
+To minimize the risk of clients getting throttled by the API, the request client SHOULD send `BatchRequestContent` instances in sequence if the `BatchRequestContentCollection` contains more than one `BatchRequestContent` instance by default. The request client MAY send `BatchRequestContent` instances in parallel by use of a configuration parameter to allow the user to opt-in to this behavior in the event that the user is aware of the potential throttling limits.
+
+## Usage Examples
+
+```cs
+// create the batch request content collection
+var batchRequestContentCollection = new BatchRequestContentCollection(graphServiceClient);
+var requestInformation = graphServiceClient.Users.ToGetRequestInformation();
+
+// add step using requestInformation
+var requestStepId = await batchRequestContentCollection.AddBatchRequestStepAsync(requestInformation);
+
+// add step using native platform HTTP request
+var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, "http://example.com");
```

Comment:
We say it should be a relative URL in the requirements. This should probably not be something "generic" but very Graph-specific in this case.